examples/pytorch_imagenet_resnet50.py

```diff
@@ -156,13 +161,25 @@ def train(epoch):
             if args.cuda:
                 data, target = data.cuda(), target.cuda()
             optimizer.zero_grad()
-            output = model(data)
-            loss = F.cross_entropy(output, target)
-            loss.backward()
+            # Split batch into sub-batches and sum gradients for all samples
+            # Note: the performance may degrade if backward_passes_per_step is
```

Comment:
We could just make it an error that throws a `ValueError` exception.  Let's make an explicit check at line ~55, and introduce `sub_batch_size` (or `backward_pass_batch_size` for clarity) at that point.